Sources/Moya/MoyaProvider.swift

```diff
@@ -25,8 +25,15 @@ public struct ProgressResponse {
     }
 }
 
+public protocol MoyaProviderProtocol: class {
```

Comment:
thoughts on `MoyaProviderProtocol` v `MoyaProviderType`? I've always preferred the latter, but I don't know if I've fallen behind the times ;p